geode-core/src/distributedTest/java/org/apache/geode/internal/cache/DeltaForceSizingFlagDUnitTest.java

```diff
@@ -0,0 +1,995 @@
+    String value;
+
+    public TestKey() {
+
+    }
+
+    public TestKey(String value) {
+      this.value = value;
+    }
+
+    @Override
+    public void fromData(DataInput in) throws IOException, ClassNotFoundException {
+      value = DataSerializer.readString(in);
+    }
+
+    @Override
+    public void toData(DataOutput out) throws IOException {
+      DataSerializer.writeString(value, out);
+    }
+
+    @Override
+    public int hashCode() {
+      final int prime = 31;
+      int result = 1;
+      result = prime * result + ((value == null) ? 0 : value.hashCode());
+      return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (this == obj) {
+        return true;
+      }
+      if (obj == null) {
+        return false;
+      }
+      if (!(obj instanceof TestKey)) {
+        return false;
+      }
+      TestKey other = (TestKey) obj;
+      if (value == null) {
+        if (other.value != null) {
+          return false;
+        }
+      } else if (!value.equals(other.value)) {
+        return false;
+      }
+      return true;
+    }
+
+  }
+
+  private static class TestObject implements DataSerializable {
+    public int sizeForSizer;
+    public int sizeForSerialization;
+
+    public TestObject() {
+
+    }
+
+
+    public TestObject(int sizeForSerialization, int sizeForSizer) {
+      super();
+      this.sizeForSizer = sizeForSizer;
+      this.sizeForSerialization = sizeForSerialization;
+    }
+
+    @Override
+    public void fromData(DataInput in) throws IOException, ClassNotFoundException {
+      sizeForSizer = in.readInt();
+      sizeForSerialization = in.readInt();
+      // We don't actually need these things.
+      in.skipBytes(sizeForSerialization);
+    }
+
+    @Override
+    public void toData(DataOutput out) throws IOException {
+      out.writeInt(sizeForSizer);
+      out.writeInt(sizeForSerialization);
+      out.write(new byte[sizeForSerialization]);
+
+    }
+
+    @Override
+    public int hashCode() {
+      final int prime = 31;
+      int result = 1;
+      result = prime * result + sizeForSerialization;
+      result = prime * result + sizeForSizer;
+      return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (this == obj) {
+        return true;
+      }
+      if (obj == null) {
+        return false;
+      }
+      if (!(obj instanceof TestObject)) {
+        return false;
+      }
+      TestObject other = (TestObject) obj;
+      if (sizeForSerialization != other.sizeForSerialization) {
+        return false;
+      }
+      if (sizeForSizer != other.sizeForSizer) {
+        return false;
+      }
+      return true;
+    }
+
+    @Override
+    public String toString() {
+      return "TestObject [sizeForSerialization=" + sizeForSerialization + ", sizeForSizer="
+          + sizeForSizer + "]";
+    }
+  }
+
+  public static class TestCacheListener extends CacheListenerAdapter {
```

Comment:
Compiler warnings can be fixed by using `TestCacheListener<K, V> extends CacheListenerAdapter<K, V>` here.